query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryRequestUtil.java

```diff
@@ -68,4 +71,23 @@ public static Expression createNullNumberLiteralExpression() {
                 .setValue(Value.newBuilder().setValueType(ValueType.NULL_NUMBER)))
         .build();
   }
+
+  public static boolean isDateTimeFunction(Expression expression) {
+    return expression.getValueCase() == ValueCase.FUNCTION
```

Comment:
Would have been niced to abstract this pinot detail (like we did with AVGRATE) before adding promql support

Reply:
Can you please elaborate more (for AVGRATE I think it was not earlier supported in qs)?

Reply:
So the concept of AVGRATE, similar to grouping by time is an abstract one that we have exposed in our own query language. In the original implementation (for both), the translation from that abstract concept to a pinot specific implementation happened in gateway which was a leak of implementation details that should be part of query service's abstraction. AVGRATE since was pushed down, so now gateway just asks for avgrate and we translate that to it's pinot equivalent (and eventually, I assume, it's promQL equivalent). Similarly, dateTimeConvert should do the same.

This may be a fair bit of work, so it's up to you guys on the approach, but what we're forcing ourselves to do right now is translating it first from our own language to a pinot expression, then from that pinot expression back to an abstraction.

Reply:
Filed https://github.com/hypertrace/query-service/issues/112